wire-protoc-integration/src/main/java/com/squareup/wire/protocwire/CodeGenerator.kt

```diff
@@ -0,0 +1,19 @@
+package com.squareup.wire.protocwire
+
+import com.squareup.wire.protocwire.Plugin.DescriptorSource
+import com.google.protobuf.compiler.PluginProtos
+
+/**
+ * Interface to be implemented by a code generator.
+ */
+interface CodeGenerator {
+    /**
+     * Generates code for the given proto file, generating one or more files to
+     * the given response.
```

Comment:
Where is the "proto file" described here? What are the roles of the 3 parameters? Is the generation to happen only in memory, can the `CodeGenerator` write on disk as well?

Reply:
I reworded the comment a bit calling out that it takes in a `.proto` file. Wdyt?